pkg/egg/parse_test.go

```diff
@@ -0,0 +1,84 @@
+package egg
+
+import (
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+func TestParse(t *testing.T) {
+	vectors := []struct {
+		file string // Test input file
+		want []types.Library
+	}{
+		{
+			file: "testdata/egg_normal",
+			want: EggNormal,
+		},
+		{
+			file: "testdata/egg_many",
+			want: EggMany,
+		},
+		{
+			file: "testdata/egg_django",
+			want: EggDjango,
+		},
+	}
+
+	for _, v := range vectors {
+		t.Run(path.Base(v.file), func(t *testing.T) {
+			f, err := os.Open(v.file)
+			require.NoError(t, err)
+			got, err := walk(f)
+			require.NoError(t, err)
+
+			assert.Equal(t, v.want, got)
+		})
+	}
+}
+
+// walk is a test helper to traverse directories and look for METADATA files
+// given that Parse only parses METADATA, as traversing directories is done in fanal
+func walk(r io.Reader) ([]types.Library, error) {
```

Comment:
this and [this](https://github.com/aquasecurity/go-dep-parser/pull/22/files?file-filters%5B%5D=.go#diff-fa141eeaea210ac278122d59e1f38d922d3ee9436afdfcc73774b64b8757f6cdR49) might not be necessary here?